accepted/00083-containerized-proxy.md

```diff
@@ -0,0 +1,87 @@
+- Feature Name: Deliver Proxy as a container
+- Start Date: 2021-10-11
+
+# Summary
+[summary]: #summary
+
+Deliver Proxy as a container application.
+
+# Motivation
+[motivation]: #motivation
+
+There are deployment scenarios in which adding a new Virtual Machine to implement Proxy functionality is cumbersome or impossible, while leveraging an existing container runtime cold be an easier option, especially in highly distributed ("Edge") environments.
+
+Having the option of deploying Proxy functionality as a container can be easier to operate (eg. deployment, independence from underlying OS, keeping up to date) and more flexible (eg. implementing active/passive HA by leveraging an existing Kubernetes environment).
+
+It is believed implementing Proxy containerization will surface a subset of common problems and solutions also useful to implement a containerized Server in future.
+
+# Detailed design
+[design]: #detailed-design
+
+## Delivery
+  - Proxy is delivered as one container
+    - Updates are delivered via new images on a public Registry
+  - Documentation is produced to describe how this container can be run on popular container runtimes: [docker](https://docs.docker.com/engine/reference/commandline/cli/) and [podman](https://podman.io/)
+  - A minimal [Helm](https://helm.sh/) chart is delivered and tested against [k3s](https://k3s.io/)
+
+## Onboarding
+  - Containerized Proxy is not a managed client, does not run `salt-minion`, is not registered in the usual way
+  - A new UI page (and API endpoint) allows to add containerized Proxies by specifying their FQDN
+    - After confirmation a System entry (in the System list) is created 
+      - Normal client actions (package management, configuration file management...) will not be available on that System
+      - Proxy-specific Server UI/API functionality, like showing the list of attached clients, will be available like on regular Proxies
+    - for each such System, the Server generates a tarball ("configuration directory") that can be downloaded
+    - Proxy container needs the configuration directory to be mounted as a volume. After start, it is fully connected and operational
+
+## Operating
+  - Container will expose Salt (4505/4506) and https (443) ports
```

Comment:
What about cobbler and network boot? I think that is expected to be supported as well for autoyast - so tftp (69/udp) exposed as well.

UEFI HTTP support https as well, but I am not sure how often will customer want to or is will to properly configure clients with CAs (I don't know how that even can be done, but it should be possible). So exposing http (80/tcp) should be considered as well. At least for specific URL path.

Reply:
Or we can have separate cobbler container? Can be viable for customers which want minimal proxy and do not network boot.
This cobbler container then can be orchestrated together with proxy container. Can be covered in separate RFC? (WDYT @SchoolGuy )

Reply:
Do we have second full cobbler installation? I was thinking that only proxy part of cobbler would be spun out to separate cobbler. Basically `/srv/tftpboot`, cgi scripts needed for tftpsync and tftp and http server.

Reply:
Well if it is just this then we already have this. And no currently we don't have a full Cobbler installation. The proxy part of Cobbler is split out since we only use a single custom wsgi script for this at the Proxy.

Reply:
Point taken about exposing ports 80 and 69.

I will open a separate discussion over the possibility of using multiple containers when the rest of this RFC is set, as I am now learning about `podman` features that might change the cost/benefit balance a bit.